### ISSUE TITLE
noisy output

### ISSUE THREAD

**Issue by yug125lk** [JuliaWolleb/Diffusion-based-Segmentation#56]
Hi, thank you again. Could you please help me? What's the impact of this condition? `--class_cond False` , 
I used it to train my custom dataset, but the model seems overfitted. How can I reduce the parameters? Could you provide me with suggestions?

**Comment by yug125lk** [comment]
When I commented out line 926 in _gaussian_diffusion.py_   and modified the rest lines (mask instead of res), the losses became equal. (custom dataset _graysclae _ I used the code to generate a grayscale image, not a segmented mask.)
`   res = torch.where(mask > 0, 1, 0)   #merge all tumor classes into one to get a binary segmentation mask`
`        mask = x_start[:, -1:, ...]
        #res = torch.where(mask > 0, 1, 0)   #merge all tumor classes into one to get a binary segmentation mask

        res_t = self.q_sample(mask, t, noise=noise)     #add noise to the segmentation channel
        x_t=x_start.float()
        x_t[:, -1:, ...]=res_t.float()
        terms = {}`

`                    model=lambda *args, r=frozen_out: r,
                    x_start=mask,
                    x_t=res_t,
                    t=t,
                    clip_denoised=False,
                )["output"]
                if self.loss_type == LossType.RESCALED_MSE:
                    # Divide by 1000 for equivalence with initial implementation.
                    # Without a factor of 1/1000, the VB term hurts the MSE term.
                    terms["vb"] *= self.num_timesteps / 1000.0

            target = {
                ModelMeanType.PREVIOUS_X: self.q_posterior_mean_variance(
                    x_start=mask, x_t=res_t, t=t
                )[0],
                ModelMeanType.START_X: mask,
                ModelMeanType.EPSILON: noise,`

`def _extract_into_tensor(arr, timesteps, broadcast_shape):
    """
    Extract values from a 1-D numpy array for a batch of indices.
    :param arr: the 1-D numpy array.
    :param timesteps: a tensor of indices into the array to extract.
    :param broadcast_shape: a larger shape of K dimensions with the batch
                            dimension equal to the length of timesteps.
    :return: a tensor of shape [batch_size, 1, ...] where the shape has K dims.
    """
    mask = th.from_numpy(arr).to(device=timesteps.device)[timesteps].float()
    while len(res.shape) < len(broadcast_shape):
        mask = mask[..., None]
    return res.expand(broadcast_shape)`
![Untitled](https://github.com/JuliaWolleb/Diffusion-based-Segmentation/assets/127154096/2d610b0f-fb98-40cc-84af-53634bc4dbc5)
